qiskit/providers/aer/backends/aerbackend.py

```diff
@@ -360,6 +360,14 @@ def _assemble_noise_model(self, circuits, **run_options):
         noise_model = run_options.get(
             'noise_model', getattr(self.options, 'noise_model', None))
 
+        # Add custom pass noise only to QuantumCircuit objects
+        if noise_model and all(isinstance(circ, QuantumCircuit) for circ in circuits):
+            npm = noise_model._pass_manager()
+            if npm is not None:
+                circuits = npm.run(circuits)
```

Comment:
Since we only have the delay pass at the moment we could improve the efficiency of this by checking if circuits have duration so we can avoid the circuit <-> dag conversion if they dont by doing:
```python
# Check if any circuits have a non-zero duration.
# Since the noise pass currently only contains delay noise we can skip it for
# zero duration circuits to avoid the overhead of circuit <-> dag conversions.
has_duration = any(circ.duration for circ in circuits)
if has_duration and npm is not None:
    circuits = npm.run(circuits)
```

Reply:
Since https://github.com/Qiskit/qiskit-terra/pull/7334 is merged in Terra, we don't need to add this hack, right?

Reply:
I know we need to care about performance very much.
But what if we have ordinary (non-scheduled) circuits with delays?
I think to do this correctly, we need to count delays in circuits efficiently.
That could be done (using `ContainsInstruction("delay")` pass) after we reconstruct the entire NoiseModel as a pass manager in the future.
I have no good idea so far, so I'll leave this line as it is.

Reply:
Does such a circuit still register as `circuit.duration = 0`? If so then yes we couldn't include this. If we need to do another pass to check, it's not worth doing  since we might as well just run the pass manager.

Reply:
I just checked this and indeed `circuit.duration` is `None` if you manually add delays, so this should be left as it currently is